codec/src/main/java/io/netty/handler/codec/MessageToByteEncoder.java

```diff
@@ -120,9 +120,9 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
                 ctx.write(msg, promise);
             }
         } catch (EncoderException e) {
-            throw e;
+            promise.setFailure(e);
         } catch (Throwable e) {
-            throw new EncoderException(e);
+            promise.tryFailure(new EncoderException(e));
```

Comment:
plz add a comment to explain why this is using `tryFailure` but above is using `setFailure`

Reply:
Actually I think using `setFailure` is better. Good catch